src/renderer/base/renderer.cpp

```diff
@@ -638,6 +638,7 @@ void Renderer::EnablePainting()
     // When the renderer is constructed, the initial viewport won't be available yet,
     // but once EnablePainting is called it should be safe to retrieve.
     _viewport = _pData->GetViewport();
+    _forceUpdateViewport = true;

```

Comment:
> (silmarillion) moving a tab to a larger window doesn't resize the control - **THIS WAS AN ATLAS BUG**

So, as it turns out, it's actually a `DxRenderer` bug¹ which just happens to mask the other bug in `Renderer`, fixed here. 😄

¹ `DxRenderer` doesn't use the `IRenderEngine::UpdateViewport` function. `UpdateViewport` was probably intended originally to tell the renderer the size of the text buffer.
It just happens to not cause any harm to Windows Terminal specifically, because it calls `IRenderEngine::SetWindowSize` and `Terminal::UserResize` while holding the console lock across both calls. So the renderer size and text buffer size always stay in sync. This isn't true for conhost though which uses an asynchronous approach to window resizing: The text buffer is resized on the window message thread, while the renderer thread is "free running" and only gets the current window size when it decides to paint. But by that point the window size might be different from the text buffer size.
This causes `DxRenderer` to crash with a SIGSEGV when you resize the OpenConsole window too quickly.